python/src/aac/lang/lsp/code_completion_provider.py

```diff
@@ -0,0 +1,89 @@
+"""Module for the Code Completion Provider which handles all code completion requests."""
+
+import os
+import logging
+from attr import Factory, attrib, attrs, validators
+
+from pygls.lsp import CompletionParams, CompletionList, CompletionItem, CompletionItemKind
+from pygls.server import LanguageServer
+from aac.parser import parse
+from aac.lang.active_context_lifecycle_manager import get_active_context
+from aac.lang.definition_helpers import get_definitions_by_root_key
+from aac.lang.definitions.definition import Definition
+
+SPACE_TRIGGER = " "
+
+
+@attrs
+class CodeCompletionProvider:
+    """Resolve various code compleetion triggers to specific code completion functions.
+
+    Attributes:
+        registered_completion_callbacks (dict[str, Callable]): The dict of trigger characters to corresponding code completion functions
+    """
+
+    registered_completion_callbacks: dict = attrib(default=Factory(dict), validator=validators.instance_of(dict))
```

Comment:
Could we just call it `completion_callbacks`? If it's in this dictionary, I'd say it's safe to assume it's "registered".